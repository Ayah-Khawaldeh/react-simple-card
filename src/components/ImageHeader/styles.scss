.card__header--image {
    position: relative;

    img {
        display: block;
        width: 100%;
        border-radius: 2px 2px 0 0;
        position: relative;
        left: 0;
        right: 0;
        top: 0;
        bottom: 0;
    }
}
