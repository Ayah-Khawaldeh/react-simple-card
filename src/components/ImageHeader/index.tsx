import React from 'react';
import classNames from 'classnames';

import './styles.scss';

export type ImageHeaderProps = {
    imageSrc: string;
    alt?: string;
    style?: any;
    className?: string;
};

export default ({ className, style, imageSrc, alt = '' }: ImageHeaderProps) => {
    return (
        <div className={classNames('card__header--image', className)} style={style}>
            <img src={imageSrc} alt={alt} />
        </div>
    );
};
