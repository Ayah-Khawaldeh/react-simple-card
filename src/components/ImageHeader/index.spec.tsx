import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ImageHeader from './';

it('renders ImageHeader correctly', () => {
    const wrapper = shallow(
        <ImageHeader alt="Alt text for the image" imageSrc="http://via.placeholder.com/350x200" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders ImageHeader correctly with className and extra styles', () => {
    const wrapper = shallow(
        <ImageHeader
            className="image-wrapper"
            style={{ padding: '3px' }}
            alt="Alt text for the image"
            imageSrc="http://via.placeholder.com/350x200"
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
});
