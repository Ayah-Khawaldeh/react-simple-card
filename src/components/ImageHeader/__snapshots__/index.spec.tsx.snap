// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageHeader correctly 1`] = `
<div
  className="card__header--image"
>
  <img
    alt=""
    src="http://via.placeholder.com/350x200"
  />
</div>
`;
